vars/piperLoadGlobalExtensions.groovy

```diff
@@ -0,0 +1,115 @@
+    'globalExtensionsRepository',
+    /** Credentials required to clone the repository*/
+    'globalExtensionsRepositoryCredentialsId',
+    /** Version of the extensions which should be used, e.g. the tag name*/
+    'globalExtensionsVersion'
+]
+
+@Field Set STEP_CONFIG_KEYS = []
+
+@Field Set PARAMETER_KEYS = [
+    /** This step will reinitialize the defaults. Make sure to pass the same customDefaults as to the step setupCommonPipelineEnvironment*/
+    'customDefaults',
+    /** This step will reinitialize the defaults. Make sure to pass the same customDefaultsFromFiles as to the step setupCommonPipelineEnvironment*/
+    'customDefaultsFromFiles'
+]
+
+/**
+ * This step is part of the step setupCommonPipelineEnvironment and should not be used outside independently in a custom pipeline.
+ * This step allows users to define extensions (https://sap.github.io/jenkins-library/extensibility/#1-extend-individual-stages) globally instead of in each repository.
+ * Instead of defining the extensions in the .pipeline folder the extensions are defined in another repository.
+ * You can also place a file called extension_configuration.yml in this repository.
+ * Configuration defined in this file will be treated as default values with a lower precedence then custom defaults defined in the project configuration.
+ * You can also define additional Jenkins libraries these extensions depend on using a yaml file called sharedLibraries.yml:
+ * Example:
+ * - name: my-extension-dependency
+ *   version: git-tag
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        def script = checkScript(this, parameters)
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        if(!configuration.globalExtensionsRepository){
+            return
+        }
+
+        dir(configuration.globalExtensionsDirectory){
+            Map gitParameters = [
+                $class: 'GitSCM',
+                userRemoteConfigs: [[url: configuration.globalExtensionsRepository]]
+            ]
+
+            if(configuration.globalExtensionsRepositoryCredentialsId){
+                gitParameters.userRemoteConfigs[0].credentialsId = configuration.globalExtensionsRepositoryCredentialsId
+            }
+
+            if(configuration.globalExtensionsVersion){
+                gitParameters.branches = [[name: configuration.globalExtensionsVersion]]
+            }
+
+            checkout(gitParameters)
+        }
+
+        String extensionConfigurationFilePath = "${configuration.globalExtensionsDirectory}/extension_configuration.yml"
```

Comment:
not checking if globalExtensionsDirectory already has a trailing slash is a bit sloppy here as it will lead to `foo//extenxion_configuration.yml`, but that should not have any real negative effects, so I would consider it ok